File: myhome-back/src/main/java/com/asdf/myhomeback/dto/AlarmRuleDTO.java
package com.asdf.myhomeback.dto;

import com.asdf.myhomeback.models.AlarmRule;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class AlarmRuleDTO {

    private Long id;
    private String rulePattern;
    private String alarmType;
    private String deviceName;

    public AlarmRuleDTO(AlarmRule alarmRule){
        this.id = alarmRule.getId();
        this.rulePattern = alarmRule.getRulePattern();
        this.alarmType = alarmRule.getAlarmType().toString();
        this.deviceName = alarmRule.getDeviceName();
    }
}

File: myhome-front/myhome-app/src/modules/admin/models/AlarmRuleDTO.ts
export interface AlarmRuleDTO {
    rulePattern: string,
    deviceName: string
}

export enum AlarmType {
    LOG,
    DEVICE
}

export interface AlarmRuleExtendedDTO extends AlarmRuleDTO {
    id: number,
    rulePattern: string;
    alarmType: AlarmType;
    deviceName: string;
}